portal-api/src/v1/validation/route-validators/route-validators.js

```diff
@@ -8,7 +11,21 @@ const bankIdValidation = param('bankId')
   .matches(/^\d+$/)
   .withMessage('The bank id provided should be a string of numbers');
 
-const mongoIdValidation = param('_id').isMongoId().withMessage("Invalid MongoDB '_id' path param provided");
+/**
+ * Validator for a path parameter which is a mongo id
+ * @param {string} paramName
+ * @returns {import('express').RequestHandler}
+ */
+const mongoIdValidation = (paramName) => (req, res, next) => {
+  const pathParam = req.params[paramName];
+  if (ObjectId.isValid(pathParam)) {
+    return next();
+  }
+  return res.status(HttpStatusCode.BadRequest).send({
+    message: `Expected path parameter '${paramName}' to be a valid mongo id`,
```

Comment:
do we want status in the body too - I think ive seen both patterns but not sure whats most common

Reply:
Have added